src/Runtime/Distributions/DiscreteChar.cs

```diff
@@ -151,6 +146,8 @@ public sealed class DiscreteChar
 
         #region Properties matching factory methods
 
+        public bool IsNull => this.storage == null;
```

Comment:
If storage can be null, then many properties and methods of this class will throw a NullReferenceException, which they are supposed to never throw.  For example, IsDigit will throw instead of returning false, GetHashCode will throw instead of returning a number, and MaxDiff will throw instead of returning infinity.  Those will all need to be fixed.  See [Property Design](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property) and [Rules for GetHashCode](https://ericlippert.com/2011/02/28/guidelines-and-rules-for-gethashcode/).

Reply:
Yeah, the `GetHashCode()` and `ToString()` should definetly be fixed.

arguably, before this proposed this code:
```
var dc = default(DiscreteChar);
var isDigit = dc.IsDigit;
```
actually throws NullReferenceException. After this change it does the same. So I don't think I break any invariants with not checking for null here.

Reply:
This change is gone

Reply:
I wanto to reiterate. That default initialized DiscreteChar thew a NullReferenceException when trying to use it. It did it my PR also. So I don't think that I introduced any harmful change in behaviour. All old code works nicely. Throwing exceptions in properties should be avoided but isn't disallowed. We do it in `Point` getter pretty much everywhere.

Anyway, I pushed a commit which introduced old behaviour. If DiscreteChar isn't assigned any distribution, it behaves like distribution created using DiscreteChar.Uniform() and never throws NullReferenctExceptions. It comes with about 1% performance penalty though in my (unscientific) tests.